domains/client/domain-executor/src/core_bundle_producer.rs

```diff
@@ -158,6 +161,16 @@ where
                             storage_proof: proof_of_election.storage_proof,
                             block_number: proof_of_election.block_number,
                             block_hash: as_core_block_hash(proof_of_election.block_hash),
+                            // TODO: override the core block info, see if there is a nicer way
+                            // later.
```

Comment:
Maybe have an enum instead of a struct so you don't need to have optional fields?